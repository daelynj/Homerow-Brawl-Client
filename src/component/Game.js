import React, { useState } from 'react';
import Board from './Board';

function Game() {
  const [words] = useState("this is text".split(" "));
  const [currentWordIndex, setCurrentWordIndex] = useState(0);
  const [currentInput, setCurrentInput] = useState("");
  const [finishTime, setFinishTime] = useState(0);
  const [start, setStart] = useState(false);

  const endGame = () => currentWordIndex < words.length ? false : true;

  function checkWord() {
    if (currentInput === words[currentWordIndex]) {
      setCurrentInput('');
      setCurrentWordIndex(currentWordIndex+1);
      return true;
    }
  }

  function getCurrentWordSubstring() {
    if (!endGame()) {
      return words[currentWordIndex].slice(0, currentInput.length);
    }
  }

  return (
    <div className="game">
      <div className="board">
        {
          <Board
            words = {words}
            currentWordSubstring = {getCurrentWordSubstring()}
            currentInput = {currentInput}
            setCurrentInput = {setCurrentInput}
            currentWordIndex = {currentWordIndex}
            setCurrentWordIndex = {setCurrentWordIndex}
            start = {start}
            setStart = {setStart}
            endGame = {endGame}
            finishTime = {finishTime}
            setFinishTime = {setFinishTime}
            checkWord = {checkWord}
        />
        }
      </div>
    </div>
  )
}

export default Game;
