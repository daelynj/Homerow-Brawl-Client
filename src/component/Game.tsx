import * as React from "react";
import { useState } from "react";
import { Board } from "./Board";

interface Props {
  updatePosition: (newPosition: any) => void;
  setCountDown: (newCountDown: any) => void;
  updateCountDown: (newCountDown: any) => void;
  raceState: any;
  ID: number;
  countDown: boolean;
}

export const Game = (props: Props) => {
  const [words] = useState<String[]>("this is text".split(" "));
  const [currentWordIndex, setCurrentWordIndex] = useState<number>(0);
  const [currentInput, setCurrentInput] = useState<string>("");
  const [finishTime, setFinishTime] = useState<number>(0);
  const [incorrectLetters, setIncorrectLetters] = useState<number>(0);
  const [countUp, setCountUp] = useState<boolean>(false);

  const endGame = () => {
    if (onFinalLetter()) {
      setCurrentWordIndex(currentWordIndex + 1);
      return true;
    }

    return currentWordIndex < words.length ? false : true;
  };

  const onFinalLetter = () => {
    if (
      currentWordIndex === words.length - 1 &&
      currentInput === words[words.length - 1]
    ) {
      return true;
    }
  };

  const getCurrentWordSubstring = () => {
    if (!endGame()) {
      return words[currentWordIndex].slice(0, currentInput.length);
    }
  };

  const checkLetter = () => {
    if (currentInput !== getCurrentWordSubstring()) {
      setIncorrectLetters(incorrectLetters + 1);
    }
    return;
  };

  return (
    <div className="game">
      <div className="board">
        {
          <Board
            updateCountDown={props.updateCountDown}
            countDown={props.countDown}
            setCountDown={props.setCountDown}
            countUp={countUp}
            setCountUp={setCountUp}
            ID={props.ID}
            raceState={props.raceState}
            updatePosition={props.updatePosition}
            words={words}
            currentWordSubstring={getCurrentWordSubstring()}
            currentInput={currentInput}
            setCurrentInput={setCurrentInput}
            currentWordIndex={currentWordIndex}
            setCurrentWordIndex={setCurrentWordIndex}
            endGame={endGame}
            finishTime={finishTime}
            setFinishTime={setFinishTime}
            checkLetter={checkLetter}
            incorrectLetters={incorrectLetters}
          />
        }
      </div>
    </div>
  );
};
