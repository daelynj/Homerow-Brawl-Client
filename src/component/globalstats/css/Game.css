.game {
  float: right;
  margin-left: 5px;
  margin-right: 5px;
  padding-top: 10px;
  padding-right: 6px;
  padding-bottom: 10px;
}

.individualhistory {
  padding: 8px;
}
