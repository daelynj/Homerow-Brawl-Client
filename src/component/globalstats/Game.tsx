import * as React from "react";
import "./css/Game.css";

interface Props {
  game: any;
}

export const Game = (props: Props) => {
  const buildDate = () => {
    let timestamp = new Date(Date.parse(props.game.created_at));
    return timestamp.toDateString();
  };
  return (
    <tbody>
      <tr>
        <th>
          <div>Date</div>
        </th>
        <th>
          <div>Time</div>
        </th>
        <th>
          <div>WPM</div>
        </th>
        <th>
          <div>Accuracy</div>
        </th>
        <th>
          <div>Mistakes</div>
        </th>
        <th>
          <div>Words</div>
        </th>
        <th>
          <div>Letters</div>
        </th>
      </tr>
      <tr>
        <td>{buildDate()}</td>
        <td>{props.game.time + " seconds"}</td>
        <td>{props.game.wpm}</td>
        <td>{props.game.accuracy + "%"}</td>
        <td>{props.game.mistakes}</td>
        <td>{props.game.words_typed}</td>
        <td>{props.game.letters_typed}</td>
      </tr>
    </tbody>
  );
};
