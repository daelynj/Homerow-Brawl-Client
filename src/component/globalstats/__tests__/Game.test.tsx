import * as React from "react";
import { Game } from "../Game";
import renderer from "react-test-renderer";

describe("Game", () => {
  let game = {
    created_at: "Monday",
    time: 3,
    wpm: 15,
    accuracy: 100,
    mistakes: 0,
    words_typed: 4,
    letters_typed: 20
  };

  it("renders a game", () => {
    const tree = renderer.create(<Game game={game} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
