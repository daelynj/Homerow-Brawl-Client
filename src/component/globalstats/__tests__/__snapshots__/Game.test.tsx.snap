// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders a game 1`] = `
<div
  className="game"
>
  <span
    className="individualhistory"
  >
    date: Monday
  </span>
  <span
    className="individualhistory"
  >
    time taken: 3 seconds
  </span>
  <span
    className="individualhistory"
  >
    WPM: 15
  </span>
  <span
    className="individualhistory"
  >
    accuracy: 100%
  </span>
  <span
    className="individualhistory"
  >
    mistakes: 0
  </span>
  <span
    className="individualhistory"
  >
    words typed: 4
  </span>
  <span
    className="individualhistory"
  >
    letters typed: 20
  </span>
</div>
`;
